pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,207 @@
+inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  constructor(
+    @inject('config.widget') protected widget: JSON
+    // This will be resolved at runtime!
+  ) {}
+  // etc...
+}
+```
+
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+## Authentication Decorator
+
+- @authenticate
+
+  Mark a controller method as requiring authenticated user, takes a strategy name 
+  as a parameter. An example is using 'BasicStrategy' to authenticate user in function
+  `whoAmI`:
+
+  ```ts
+  // my-controller.ts
+  import { authenticate } from '@loopback/authentication';
+  import { inject } from '@loopback/context';
+
+  class MyController {
+    constructor(
+      @inject(BindingKeys.Authentication.CURRENT_USER)
+      private user: UserProfile,
+    ) {}
+
+    @authenticate('BasicStrategy')
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+### Model decorators
+
+- `@model`
+
+  Define a model in a repository, see [define-models](Repositories.html#define-models)
+
+- `@property`
+
+  Define a property in a model, see [define-models](Repositories.html#define-models)
+
+### Relation decorators
+
+  *This feature has not yet been released in alpha form. Documentation will be* 
+  *added here as this feature progresses.*
+
+- `@relation`
+
+  Register a general relation
+
+- `@belongsTo` `@hasOne` `@hasMany` `@embedsOne` `@embedsMany` `@referencesOne` `@referencesMany`
+
+  Register a specific relation
+
+### Repository decorator
+
+- `@repository`
+
+  A repository represents a collection of data and the means to access that data, 
+  the decorator either injects an existing repository or creates a repository 
+  from a model and a datasource.
+
+  The injection example can be found in [Repository#controller-configuration](Repositories.html#controller-configuration)
+
+  ```ts
+  // my-controller.ts
+  // Know more about how to create model 'Todo' and datasource 'datasource', 
+  // please see example in [Thinking in LoopBack](Thinking-in-LoopBack.htm#define-product-model-repository-and-data-source)
+  import { Todo } from '{path_of_Todo_model}.ts';
+  import { datasource } from '{path_of_datasource}.ts';
+
+  export class TodoController {
+    @repository(Todo, datasource)
+    repository: EntityCrudRepository<Todo, number>;
+    ... ...
+  }
+  ```
+  
+  If the model or datasource is already bound to the app, you can create the 
+  repository by providing their names instead of the classes. For example:
+
+  ```ts
+  // with `datasource` and `Todo` already defined.
+  app.bind('datasources.ds').to(datasource);
+  app.bind('repositories.todo').to(Todo);
+
+  export class TodoController {
+    @repository('todo', 'ds')
+    repository: EntityCrudRepository<Todo, number>;
+    ... ...
```

Comment:
Replace with:
```
 // etc
```

This may seem trivial, but it lets people paste this into an editor with less friction.

They're still going to see errors due to things not being defined, such as `app`, but if they're following along with our examples, they might already have the rest of the groundwork laid out, which would mean the snippet could be useful.